### DISCUSSION TITLE
Adding custom class

### DISCUSSION THREAD

**Discussion by kmarler18** [tailwindlabs/tailwindcss#10067] [Help] [answered]
Hello All!

My project requires a box-shadow with these specific properties...

`box-shadow: 0 -2px 5px 0 rgba(0,0,0,0.09)`

While Tailwind offers close approximations, I'd like to use the specific properties above. I'd like to name this class "shadow". Tailwind already has a "[shadow](https://tailwindcss.com/docs/box-shadow)" class. After reading the [docs](https://tailwindcss.com/docs/adding-custom-styles), I figured the best way to override it, would be to add the following code to my main css file...

```
@layer utilities {
        .shadow {
         box-shadow: 0 -2px 5px 0 rgba(0,0,0,0.09);
         }
}
```

I'd like to get your thoughts on if there is a better way to go about this process or if I'm on the right track.

Thanks a million.

From the guy new to Tailwind,
Kyle

**Comment by andymccray** [comment, marked as answer]
Hey @kmarler18 👋 

That's a fine solution but there are a [couple of other ways](https://tailwindcss.com/docs/box-shadow#using-custom-values) you could go which will offer more flexibility (hover states, responsive variants etc.) and will avoid the need to add an explicit CSS rule.

If this is a class you’re wanting to use in a bunch of places you can add it to the `tailwind.config.js` file as an additional box shadow option:

```js
module.exports = {
  theme: {
    extend: {
      boxShadow: {
        'custom': '0 -2px 5px 0 rgba(0,0,0,0.09)',
      }
    }
  }
}
```

You can then use the `shadow-custom` class anywhere in your project (or call it anything else that makes sense).

If this is a one-off style you can create an arbitrary CSS rule on the fly:

```html
<div class="shadow-[0_-2px_5px_0_rgba(0,0,0,0.09)]"> ... </div>
```

Hope that helps!

**Reply by kmarler18** [reply]
Hey @andymccray,

Thanks for your input! That was really helpful!

I will say though, unless I'm missing something in the [docs](https://tailwindcss.com/docs/adding-custom-styles#using-modifiers-with-custom-css), I think you can still use modifiers like `hover:` even with the technique I originally employed.

To that end, both techniques appear even more similar, though looking at the Developer Tools, applying the original technique doesn't change the CSS variable `--tw-shadow` whereas using your proposed technique does manipulate that value.

![Screen Shot 2022-12-13 at 6 15 30 PM](https://user-images.githubusercontent.com/96091818/207466307-dee913b7-058f-473b-958b-16be7994d3c0.png)

![Screen Shot 2022-12-13 at 6 16 25 PM](https://user-images.githubusercontent.com/96091818/207466316-e431ee3d-c1e3-41de-992c-ef576a663f4c.png)

But, now I'm just talking to myself. You answered my question! Well done!
